clang/test/CodeGen/cheri/hybrid-decay.c

```diff
@@ -13,6 +13,10 @@ struct S {
 // and, if it generates erroneous addrspacecast's, those somehow persist,
 // whereas the other two variants, and all the decay-only tests, all have their
 // addrspacecast's folded if optimisations are enabled, masking the bug.
+//
+// The variants that implicitly cast to void * __capability also used to
+// previously crash Clang only when compiling C++, but are included for
+// completeness here too.
```

Comment:
Do we compile these tests as C++? I don't see them in the .cpp file?

Reply:
Seems that was an oversight in the original set of tests; the C++ ones just test C++ things, and the C ones test C things, yet nothing tests the C compiled as C++...